common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -0,0 +1,834 @@
+                errorResult[0] = error;
+                latch.countDown();
+            }
+        });
+
+        // Wait for the async op to complete...
+        try {
+            latch.await();
+
+            if (null != result[0]) {
+                return result[0];
+            } else {
+                throw errorResult[0];
+            }
+        } catch (final InterruptedException e) {
+            Logger.error(
+                    TAG,
+                    "Interrupted while waiting on callback.",
+                    e
+            );
+
+            throw new ClientException(
+                    INTERRUPTED_OPERATION,
+                    e.getMessage(),
+                    e
+            );
+        }
+    }
+
+    @Override
+    public void getRequestConfirmation(@NonNull final TaskCompletedCallbackWithError<String, ClientException> callback) {
+        sThreadExecutor.submit(new Runnable() {
+            @Override
+            public void run() {
+                // Vars for error handling...
+                final Exception exception;
+                final String errCode;
+
+                try {
+                    final KeyStore.Entry keyEntry = mKeyStore.getEntry(KEYSTORE_ENTRY_ALIAS, null);
+                    final KeyPair rsaKeyPair = getKeyPairForEntry(keyEntry);
+                    final RSAKey rsaKey = getRsaKeyForKeyPair(rsaKeyPair);
+                    final String base64UrlEncodedJwkJsonStr = getReqCnfForRsaKey(rsaKey);
+
+                    callback.onTaskCompleted(base64UrlEncodedJwkJsonStr);
+
+                    // We're done.
+                    return;
+                } catch (final KeyStoreException e) {
+                    exception = e;
+                    errCode = KEYSTORE_NOT_INITIALIZED;
+                } catch (final NoSuchAlgorithmException e) {
+                    exception = e;
+                    errCode = NO_SUCH_ALGORITHM;
+                } catch (final UnrecoverableEntryException e) {
+                    exception = e;
+                    errCode = INVALID_PROTECTION_PARAMS;
+                } catch (final JOSEException e) {
+                    exception = e;
+                    errCode = THUMBPRINT_COMPUTATION_FAILURE;
+                } catch (final JSONException e) {
+                    exception = e;
+                    errCode = JSON_CONSTRUCTION_FAILED;
+                }
+
+                final ClientException clientException = new ClientException(
+                        errCode,
+                        exception.getMessage(),
+                        exception
+                );
+
+                Logger.error(
+                        TAG,
+                        clientException.getMessage(),
+                        clientException
+                );
+
+                callback.onError(clientException);
+            }
+        });
+    }
+
+    @Override
+    public String getAuthorizationHeaderValue(@NonNull final String httpMethod,
```

Comment:
Would the 'minted access token' or anything along the line be easier/clearer to understand?

Reply:
Resolved in a43fdacb667e821ffe22be05c8fd9e97ffd201ea